### PR TITLE
Documentation: change installation instruction about source code

### PR DESCRIPTION
**Pull request by AroosaIjaz**
**Description of the Change:**

1. Adding detailed commands for installation from source code like we discussed in an earlier PR. 

2. Adding the discussion forum  as a direct link. 

**Benefits:**
Having a direct visible help tab called `Get Help` in the side gallery is easier to spot. It does not depend on the user scrolling down the front page or going through all the side tabs to find the page on `Support`. I personally always look for the `Help` tab on a website when I have issues. We can also rename it to `Discussion Forum `. What do you guys think ?

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -134,6 +134,7 @@ PennyLane is **free** and **open source**, released under the Apache License, Ve
    installing
    plugins
    research
+   Get Help<https://discuss.pennylane.ai/>
```
Comment (co9olguy) [inline doc/index.rst]:
```suggestion
   Get help<https://discuss.pennylane.ai/>
```

Comment (co9olguy) [reply doc/index.rst]: Alternative suggestions:
`Help`
`Help forums`
@josh146 any preferences?

Comment (josh146) [reply doc/index.rst]: `Questions and answers`
`Discussion`
`Ask for help`

Comment (josh146) [reply doc/index.rst]: I quite like `Ask for help`?

Comment (AroosaIjaz) [reply doc/index.rst]: Just `Help` or `Ask for help` are good.

**Comment by co9olguy** [comment]
I'm going to merge this into the `doc-refactor` branch. We can revisit them name later, but atm I'm liking "Get help" the best :)

**Pull request merged**